pkg/controllers/infracluster/infracluster_controller_test.go

```diff
@@ -0,0 +1,266 @@
+				Expect(cl.Get(ctx, types.NamespacedName{Namespace: actual.Namespace, Name: actual.Name}, actual)).To(Succeed())
+				Expect(actual.Annotations).To(HaveKeyWithValue(clusterv1.ManagedByAnnotation, thirdPartyAnnotation))
+				Expect(actual.Status.Ready).To(BeFalse(), "should have not changed the Status.Ready field")
+			})
+		})
+	})
+
+	Context("When there is an InfraCluster with an externally ManagedBy Annotation with cluster-capi-operator value", func() { //nolint:dupl
+		var ic *awsv1.AWSCluster
+		BeforeEach(func() {
+			ic = infraClusterWithExternallyManagedByAnnotation.DeepCopy()
+			Expect(cl.Create(ctx, ic)).To(Succeed())
+		})
+		AfterEach(func() {
+			Expect(test.CleanupAndWait(ctx, cl, ic)).To(Succeed())
+		})
+
+		Context("When the InfraCluster is Ready", func() {
+			BeforeEach(func() {
+				mustReconcile(ocpInfraAWS, ic, true)
+			})
+
+			It("should not change the Status.Ready field", func() {
+				actual := &awsv1.AWSCluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      defaultInfraClusterName,
+						Namespace: defaultCAPINamespace,
+					},
+				}
+
+				Expect(cl.Get(ctx, types.NamespacedName{Namespace: actual.Namespace, Name: actual.Name}, actual)).To(Succeed())
+				Expect(actual.Annotations).To(HaveKeyWithValue(clusterv1.ManagedByAnnotation, managedByAnnotationValueClusterCAPIOperator))
+				Expect(actual.Status.Ready).To(BeTrue(),
+					"should have not changed the Status.Ready field")
+			})
+		})
+		Context("When the InfraCluster is not Ready", func() {
+			BeforeEach(func() {
+				mustReconcile(ocpInfraAWS, ic, false)
+			})
+
+			It("should change the Status.Ready field to true", func() {
+				actual := &awsv1.AWSCluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      defaultInfraClusterName,
+						Namespace: defaultCAPINamespace,
+					},
+				}
+
+				Expect(cl.Get(ctx, types.NamespacedName{Namespace: actual.Namespace, Name: actual.Name}, actual)).To(Succeed())
+				Expect(actual.Annotations).To(HaveKeyWithValue(clusterv1.ManagedByAnnotation, managedByAnnotationValueClusterCAPIOperator))
+				Expect(actual.Status.Ready).To(BeTrue(),
+					"should have changed the Status.Ready field to true")
+			})
+		})
+	})
+
+})
+
+func mustPatchAWSInfraClusterReadiness(awsInfraCluster *awsv1.AWSCluster, readiness bool) {
+	awsInfraClusterCopy := awsInfraCluster.DeepCopy()
+	patchBase := client.MergeFrom(awsInfraClusterCopy.DeepCopy())
+	awsInfraClusterCopy.Status = awsv1.AWSClusterStatus{
+		Ready: readiness,
+	}
+	Expect(cl.Status().Patch(ctx, awsInfraClusterCopy, patchBase))
+}
+
+func mustReconcile(ocpInfra configv1.Infrastructure, awsInfraCluster *awsv1.AWSCluster, readiness bool) {
+	mustPatchAWSInfraClusterReadiness(awsInfraCluster, readiness)
+
+	r := &InfraClusterController{
+		ClusterOperatorStatusClient: operatorstatus.ClusterOperatorStatusClient{
+			Client: cl,
+		},
+		Infra:    ocpInfra,
+		Platform: ocpInfra.Status.PlatformStatus.Type,
+	}
+
+	_, err := r.Reconcile(ctx, reconcile.Request{
```

Comment:
Why are we calling reconcile manually rather than setting it up with the manager and using asynchronous assertions?
We've seen in the past that single reconciles can be fragile as there's no retry logic. Prefer to run the manager for each test and trigger the reconcile via the watches

Reply:
Switched to using the manager.